### PR TITLE
PHPLIB-918: Add test that reads are not retried in a transaction

### PR DESCRIPTION
**Pull request by levon80999**
https://jira.mongodb.org/browse/PHPLIB-918

Synced with https://github.com/mongodb/specifications/commit/c5d1c778dba6439de72e6a3dbbaeab9de7539306

### CODE REVIEW

File: tests/UnifiedSpecTests/transactions/retryable-abort-handshake.json [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+          "_id": 1,
+          "x": 11
+        }
+      ]
+    }
+  ],
+  "tests": [
+    {
+      "description": "AbortTransaction succeeds after handshake network error",
+      "skipReason": "DRIVERS-2032: Pinned servers need to be checked if they are still selectable",
```
Comment (jmikola) [inline tests/UnifiedSpecTests/transactions/retryable-abort-handshake.json:69]: I noted from the downstream changes in [DRIVERS-746](https://jira.mongodb.org/browse/DRIVERS-746) that both of the "retryable" tests are skipped, so it was not necessary to add these to `UnifiedSpecTest::$incompleteTests`.

**Review by jmikola — CHANGES_REQUESTED** [review]
For consistency in the future, please use `PHPLIB-###: ` as a commit prefix instead of wrapping the issue in brackets. I revised the PR title, which should be sufficient to ensure the correct message is used when squash-merging.

The commit message is missing a comment indicating the [mongodb/specifications](https://github.com/mongodb/specifications) commit to which you've synced. When we discussed this earlier, I referenced the commits in https://github.com/mongodb/mongo-php-library/pull/961 as an example. Please add the following to your commit message:

```
Synced with mongodb/specifications@<full commit hash>
```

GitHub will then render that as a short link in its UI, as in https://github.com/mongodb/mongo-php-library/commit/f8eff8dd5b039ea34b1379907d76ca47e5113250.

Feel free to amend your commit and force-push to your PR branch.

File: tests/UnifiedSpecTests/transactions/retryable-abort-handshake.json (hunk 1 above)
Comment (levon80999) [reply tests/UnifiedSpecTests/transactions/retryable-abort-handshake.json:69]: I noted it too and that is why I didn't add new lines in UnifiedSpecTest::$incompleteTests. Also I double checked the [DRIVERS-2032](https://jira.mongodb.org/browse/DRIVERS-2032) ticket which is the parent ticket of [DRIVERS-746](https://jira.mongodb.org/browse/DRIVERS-746).

**Review by jmikola — APPROVED** [review]
CI failures are unrelated. I believe they are actually both fixed in master, but your PR branch is based on an older master commit. For example, the [tests/UnifiedSpecTests/change-streams](https://github.com/mongodb/mongo-php-library/tree/03735ac5841e18eecc4e77510c05960224152af7/tests/UnifiedSpecTests/change-streams) directory corresponding to this PR's commit doesn't show any recent modifications to the two failing change stream tests, which were recently updated in https://github.com/mongodb/mongo-php-library/commit/a292d0786a3fd31e3b455624df6da956db23ab46 and https://github.com/mongodb/mongo-php-library/commit/f8eff8dd5b039ea34b1379907d76ca47e5113250. For future reference, I suggest rebasing your PR branch on master regularly if you're already comfortable with force-pushing. It will limit conflicts (granted, that's not an issue here) and help avoid unrelated CI failures that might have been recently addressed.

Please just remove the full issue URL from the commit message when squash-merging, as I mentioned in https://github.com/mongodb/mongo-php-library/pull/967#issuecomment-1234570317.

**Comment by jmikola** [comment]
@levon80999: FYI, the merge commit (https://github.com/mongodb/mongo-php-library/commit/4bc4607a185db5f52ea3a24dbe36cd25112d00e8) was missing the following:

```
Synced with mongodb/specifications/commit/c5d1c778dba6439de72e6a3dbbaeab9de7539306
```

**Pull request merged**